src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -59,11 +62,41 @@ internal TableServiceClient TableClient
         {
             get
             {
-                if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null && !_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableClient))
+                if (_tableClient is null && !_environment.IsPlaceholderModeEnabled())
                 {
-                    _logger.LogWarning("An error occurred initializing the Table Storage Client. We are unable to record diagnostic events, so the diagnostic logging service is being stopped.");
-                    _isEnabled = false;
-                    StopTimer();
+                    if (!_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableClient))
+                    {
+                        DisableService();
+                        Logger.ServiceDisabledFailedToCreateClient(_logger);
+                        return _tableClient;
+                    }
+
+                    try
+                    {
+                        // When using RBAC, we need "Storage Table Data Contributor" as we require to list, create and delete tables and query/insert/delete entities.
```

Comment:
Is one of the log entries we log when this fails include this information (""Storage Table Data Contributor"" role is needed)?

Reply:
Will these logs go to AI where customer can see it?

Reply:
Not right as we're not currently emitting warnings to App Insights and I feel warning is the right log level for this scenario. They will start flowing when we enable that. We're not explicitly mentioning the role in the log message, but we can totally do that:

https://github.com/Azure/azure-functions-host/blob/34ed87ea3786aa97c1440bcd92fbc0bca300a529/src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.Log.cs#L23

Reply:
Added the role in the log message.